tests/test_printallocations.py

```diff
@@ -0,0 +1,116 @@
+
+import unittest
+import sys
+
+from models.dojo import Dojo
+
+
+class TestPrintAllocation(unittest.TestCase):
+    """It tests for the functionality print room."""
+
+    allocation_office = "green"\
+                        "\n------------------------------\n"\
+                        "[DAMILOLA DURODOLA]"
+
+    allocation_living = "blue"\
+                        "\n------------------------------\n"\
+                        "[KOYA GBOYE]"
+
+    def setUp(self):
+        """sets."""
+        self.dojo = Dojo()
+        self.dojo.office_rooms = {}
+        self.dojo.living_rooms = {}
+        self.dojo.persons = {}
+
+    def test_print_allocations_for_office(self):
+        """."""
+        room_detail = {
+            '<room_type>': 'office',
+            '<room_name>': ['green']
+            }
+        person_details = {
+            '<first_name>': 'Damilola',
+            '<last_name>': 'Durodola',
+            '<FELLOW/STAFF>': 'staff',
+            '<wants_accomodation>': ''
+        }
+        allocate_file = {
+            '--o': None
+        }
+        self.dojo.create_room(room_detail)
+        self.dojo.add_person(person_details)
+        self.dojo.print_allocations(allocate_file)
+        result = "\n".join(sys.stdout.getvalue().split("\n")[3:6])
```

Comment:
Add a comment  explaining what is happening here or better yet, break up the function.